chasm/tree.go

```diff
@@ -919,13 +990,39 @@ func unmarshalProto(
 	return value, nil
 }
 
-// Ref implements the CHASM Context interface
-func (n *Node) Ref(
+// RefC implements the CHASM Context interface
+func (n *Node) RefC(
 	component Component,
 ) (ComponentRef, bool) {
-	// TODO: Implement this method.
-	// Currently returning an empty reference to unblock tests.
-	return ComponentRef{}, true
+	// TODO: return error
+	_ = n.syncSubComponents()
```

Comment:
logging the error should be good enough I guess. The same error will be encountered when closing transaction and get returned?

Reply:
But yeah returning error is also a good idea and we can define an ErrComponentNotFound constant.